[OSDOCS-4774] ShiftStack 4.12 release notes (#54280)
openshift/openshift-docs: release_notes/ocp-4-12-release-notes.adoc

For more information, see xref:../networking/network_observability/network-observability-overview.html#network-observability-overview[Network Observability].

[id="ocp-4-12-nw-shiftstack-ipv6-secondary"]
==== IPv6 for secondary network interfaces on {rh-openstack}

IPv6 for secondary network intefaces is now supported in clusters that run on {rh-openstack}. 

// TODO: For more information,see ${DOC_WHEN_MERGED} 

[id="ocp-4-12-nw-shiftstack-udp-load-balancing"]
==== UDP support for load balancers on {rh-openstack}

> UDP is now supported for load balancers for clusters that run on {rh-openstack}.

Reviewer: I'd rephrase it into "UDP protocol". But anyway in general we should say that OpenStack is now using the external cloud provider (cloud-controller-manager/CCM). This will tell way more for people actually knowing about Kubernetes. UDP LBs support is just one of the features we get by using CCM. Another one is e.g. ability to use OVN Octavia provider.

Author: ```suggestion
Resulting from the switch to an external cloud provider, UDP is now supported for load balancers for clusters that run on {rh-openstack}.
```

@dulek Am I wrong, or is the "P" in "UDP" for "Protocol?"

For more information, see xref:../networking/network_observability/network-observability-overview.html#network-observability-overview[Network Observability].

[id="ocp-4-12-nw-shiftstack-ipv6-secondary"]
==== IPv6 for secondary network interfaces on {rh-openstack}

> IPv6 for secondary network intefaces is now supported in clusters that run on {rh-openstack}. 

Reviewer 2: ```suggestion
IPv6 for secondary network interfaces is now supported in clusters that run on {rh-openstack}. 
```

// TODO: For more information,see ${DOC_WHEN_MERGED} 

[id="ocp-4-12-nw-shiftstack-udp-load-balancing"]
==== UDP support for load balancers on {rh-openstack}

> Resulting from the switch to an external {rh-openstack} cloud provider, UDP is now supported for load balancers for clusters that run on that platform.

Reviewer 2: ```suggestion
Resulting from the switch to an external OpenStack cloud provider, UDP is now supported for load balancer type services for clusters that run on that platform.
```

Author: @eurijon I find this phrasing a little hard to parse. As an alternative, would you prefer "services that have the load balancer type" or "`LoadBalancer` services" here?

Reviewer 2: +1 for "LoadBalancer services" :)

Author: ```suggestion
Resulting from the switch to an external OpenStack cloud provider, UDP is now supported for `LoadBalancer` services for clusters that run on that platform.
```

For more information, see xref:../networking/network_observability/network-observability-overview.html#network-observability-overview[Network Observability].

[id="ocp-4-12-nw-shiftstack-ipv6-secondary"]
==== IPv6 for secondary network interfaces on {rh-openstack}

IPv6 for secondary network interfaces is now supported in clusters that run on {rh-openstack}. 

> // TODO: For more information,see ${DOC_WHEN_MERGED} 

Author: ```suggestion
For more information,see xref:../post_installation_configuration/network-configuration.adoc#nw-osp-pod-connections-ipv6_post-install-network-configuration[Enabling IPv6 connectivity to pods on {rh-openstack}].
```

While these hardware models remain fully supported in {product-title} 4.12, Red Hat recommends that you use later hardware models.

[id="ocp-4-12-ne-kuryr"]
==== Kuryr support for clusters that run on {rh-openstack}

> In {product-title} 4.12, support for Kuryr on clusters that run on {rh-openstack} is deprecated. In an upcoming release, support will be removed.

Reviewer: "In an upcoming release" suggests it can be 4.13 and we aim for removal to happen no earlier than 4.14.

Author: Should be addressed by https://github.com/openshift/openshift-docs/pull/54280/commits/0000456c104ec8b89042a487f75370979fa4a506

Reviewer: Yup, it is.

* When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])

> * Fully populated load balancers on {rh-openstack} can be stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]

Reviewer 3: ```suggestion
* Fully populated load balancers on {rh-openstack} created with OVN driver can contain pool stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]
```

> * Bulk member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]

Reviewer 3: ```suggestion
* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
```

> 

Reviewer 4: There's [a known issue](https://issues.redhat.com/browse/OCPBUGS-5036) where the external cloud provider doesn't pick up the updated credentials after a rotation. A workaround exists that consists in killing the pod for them to consume the new credential. It could be problematic for customers on OpenStack where the external cloud provider has replaced the in-tree one.

Author: Is this specific to ShiftStack, or does it impact other platforms, too?

Reviewer 4: All platforms using external cloud provider. I can't find the list of what are these platforms in 4.12.

Reviewer 3: It looks like this is the [list](https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/release-4.12/vendor/github.com/openshift/library-go/pkg/cloudprovider/external.go#L35-L40) of platforms using external cloud provider in 4.12 by default.

Author: See https://github.com/openshift/openshift-docs/pull/54280/commits/7d5e3a7cfcc39dc8a17d6d8ca4d39b1374173fa9

Reviewer 4: Aren't we missing OpenStack in the list of affected platforms?

Author: Added. :)

> * Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]

Reviewer: This only affects OSP17, should we say that?

Author: @dulek Both, or just L2163? 

```suggestion
* Fully populated load balancers on {rh-openstack} 17 that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]

* Bulk load-balancer member updates on {rh-openstack} 17 can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
```

Reviewer: Aww, no, I'm wrong, it's all good as is. Sorry, so many bugs I got confused, this comment thread can be disregarded.

* Fully populated load balancers on {rh-openstack} that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*])

* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])

* Clusters that use external cloud providers can fail to retrieve updated credentials after rotation. The following platforms are affected:
+
--
** {alibaba}
** IBM Cloud VPC
** IBM Power
** {VirtProductName}
** {rh-openstack}
--
+
As a workaround, restart openshift-cloud-controller-manager pods by running the following command:
+
[source,terminal]
----
$ oc delete pods --all -n openshift-cloud-controller-manager

> ----

Author: FYI @imatza-rh

Reviewer 5: The W/A looks good. Thanks

Author: TY!

* When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])

> * Fully populated load balancers on {rh-openstack} that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*])

Reviewer 2: @MaysaMacedo @dulek this issue has been solved in the latest z-streams, right? so this is here in case a customer is using a z-stream without the fix?

Reviewer 3: @eurijon yes

> * Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])

Reviewer 6: Maybe markup `PUT`

Author: I can find both styles throughout our docs, and nothing really definitive in the guidelines. Going to leave this as it is with v. little confidence. 

No. I'm going to change it. Again, though, very little confidence. Very much a gut thing. :shrug: 

```suggestion
* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to `PUT` requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])
```

> * Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to `PUT` requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])

Reviewer 2: @MaysaMacedo @dulek same here

Reviewer 3: yes

> * Clusters that use external cloud providers can fail to retrieve updated credentials after rotation. The following platforms are affected:

Reviewer 3: @JoelSpeed @mandre Can you confirm if the listed platforms are affected by [this bug](https://issues.redhat.com/browse/OCPBUGS-5036)?

Reviewer 7: To my knowledge, this list is correct for 4.12 yes

+
--
** {alibaba}
** IBM Cloud VPC
** IBM Power
** {VirtProductName}
** {rh-openstack}
--
+

> As a workaround, restart openshift-cloud-controller-manager pods by running the following command:

Reviewer 6: Markup `openshift-cloud-controller-manager`

Author: ```suggestion
As a workaround, restart `openshift-cloud-controller-manager` pods by running the following command:
```

+
[source,terminal]
----
$ oc delete pods --all -n openshift-cloud-controller-manager
----
+
(link:https://issues.redhat.com/browse/OCPBUGS-5036[*OCPBUGS-5036*])

> * There is a known issue when cloud-provider-openstack tries to create health monitors on OVN load balancers by using the API to create fully populated load balancers. These health monitors become stuck in a `PENDING_CREATE` status. After their deletion, associated load balancers are are stuck in a `PENDING_UPDATE` status. There is no workaround. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2143732[*BZ#2143732*])

Reviewer 6: Should `cloud-provider-openstack` be marked up? 🤷

Author: Arguably. Gonna go with it. 

```suggestion
* There is a known issue when `cloud-provider-openstack` tries to create health monitors on OVN load balancers by using the API to create fully populated load balancers. These health monitors become stuck in a `PENDING_CREATE` status. After their deletion, associated load balancers are are stuck in a `PENDING_UPDATE` status. There is no workaround. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2143732[*BZ#2143732*])
```
